tests/test_interrupted_sync.py

```diff
@@ -0,0 +1,282 @@
+                        in first_sync_records.get(stream, {}).get('messages', [])
+                        if record.get('action') == 'upsert'
+                        and record.get('data').get('owner_resource') == 'shop']
+
+                if stream != 'metafields':
+                    resuming_sync_messages = [
+                        record.get('data') for record
+                        in resuming_sync_records.get(stream, {}).get('messages', [])
+                        if record.get('action') == 'upsert']
+                else:
+                    resuming_sync_messages = [
+                        record.get('data') for record
+                        in resuming_sync_records.get(stream, {}).get('messages', [])
+                        if record.get('action') == 'upsert'
+                        and record.get('data').get('owner_resource') == 'shop']
+
+                first_bookmark_value = first_sync_state.get('bookmarks', {}).get(stream, {})
+                first_bookmark_value = list(first_bookmark_value.values())[0]
+                resuming_bookmark_value = resuming_sync_state.get('bookmarks', {}).get(stream, {})
+                resuming_bookmark_value = list(resuming_bookmark_value.values())[0]
+
+                replication_key = next(iter(expected_replication_keys[stream]))
+                first_bookmark_value_utc = self.convert_state_to_utc(first_bookmark_value)
+                resuming_bookmark_value_utc = self.convert_state_to_utc(resuming_bookmark_value)
+                if stream in new_state['bookmarks'].keys():
+                    simulated_bookmark = new_state['bookmarks'][stream]
+                else:
+                    simulated_bookmark = None
+                if stream in new_state['bookmarks'].keys():
+                    simulated_bookmark_value = list(simulated_bookmark.values())[0]
+                else:
+                    simulated_bookmark_value = None
+
+                youngest_first_sync_date = max(
+                    self.parse_date(record.get(replication_key))
+                    for record in first_sync_messages)
+
+                if len(resuming_sync_messages) > 0:  # 'collects' stream may be empty
+                    actual_oldest_resuming_replication_date = min(
+                        self.parse_date(record.get(replication_key))
+                        for record in resuming_sync_messages)
+
+                if stream in new_state['bookmarks'].keys():
+                    expected_resuming_sync_start_time = self.parse_date(first_bookmark_value)
+                else:
+                    # For not yet started streams there will be no bookmark and we should
+                    # sync all records the from the beginning of the original sync.
+                    expected_resuming_sync_start_time = min(
+                        self.parse_date(record.get(replication_key))
+                        for record in first_sync_messages)
```

Comment:
This can also be merged with those other `if`s, but it's starting to feel like the test could use some rearchitecting

Reply:
merged.